include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h

```diff
@@ -45,6 +45,25 @@ __create_accessor(_BufferType& __buf, _DiffType __offset, _DiffType __n)
 
     return {__buf, sycl::range<1>(__n_acc), __offset};
 }
+
+// Evaluates to true if the provided type is an iterator with a value_type and if the implementation of a
+//  std::vector<value_type>::iterator can be distinguished from std::vector<value_type, usm_allocator>::iterator
```

Comment:
It makes sense to add some explanation in the doc about that fact.
Lets create a reference issue.